frontend/src/bundles/common/components/select/select.tsx

```diff
@@ -0,0 +1,113 @@
+import {
+    type Control,
+    type FieldErrors,
+    type FieldPath,
+    type FieldValues,
+} from 'react-hook-form';
+import { type GroupBase, type Props } from 'react-select';
+import ReactSelect from 'react-select';
+import makeAnimated from 'react-select/animated';
+
+import {
+    useCallback,
+    useFormController,
+} from '~/bundles/common/hooks/hooks.js';
+
+import { DropdownIndicator } from './libs/components/dropdown-indicator.js';
+import { getStyles } from './libs/styles/styles.js';
+import { type SelectOption, type SelectValue } from './libs/types/types.js';
+
+type Properties<
+    T extends FieldValues,
+    IsMulti extends boolean = false,
+    Group extends GroupBase<SelectOption> = GroupBase<SelectOption>,
+> = Props<SelectOption, IsMulti, Group> & {
+    name: FieldPath<T>;
+    control: Control<T, null>;
+    errors: FieldErrors<T>;
+    label?: string;
+};
+
+const animatedComponents = makeAnimated();
+
+const Select = <
+    T extends FieldValues,
+    IsMulti extends boolean = false,
+    Group extends GroupBase<SelectOption> = GroupBase<SelectOption>,
+>({
+    name,
+    styles,
+    control,
+    errors,
+    options,
+    placeholder = 'Select something...',
+    label = '',
+    isMulti,
+    isDisabled = false,
+    isClearable = false,
+    isSearchable = false,
+    ...rest
+}: Properties<T, IsMulti, Group>): JSX.Element => {
+    const { field } = useFormController({ name, control });
+    const error = errors[name]?.message as string;
+
+    const handleSelectValue = (
+        value: SelectValue | SelectValue[],
+    ): SelectOption | SelectOption[] | undefined => {
+        if (isMulti && value) {
+            return (options as SelectOption[]).filter((option) =>
+                (value as SelectValue[]).includes(option.value),
+            );
+        } else if (!isMulti && value) {
+            return (options as SelectOption[]).find(
+                (option) => option.value === value,
+            );
+        }
+    };
+
+    const handleChange = useCallback(
+        (selectedOptions: unknown): void => {
```

Comment:
Maybe instead of `unknown` it is better to explicitly specify `SelectOption | SelectOption[]` type?